spec.bs

```diff
@@ -465,6 +468,13 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
   |config| and true.
 1. If |auctionConfig| is failure, then [=exception/throw=] a {{TypeError}}.
 1. Let |p| be [=a new promise=].
+1. Let |configMapping| be |global|'s [=associated Document=]'s [=traversable navigable=]'s
+  [=traversable navigable/fenced frame config mapping=].
+1. Let |pendingConfig| be the result of [=constructing a pending fenced frame config=] with
+  |auctionConfig|.
+1. Let |urn| be the result of running [=fenced frame config mapping/store a pending config=] on
+  |configMapping| with |pendingConfig|.
+1. If |urn| is failure, then [=reject=] |p| and return |p|.

```

Comment:
Remember to pass a _reason_ to [=reject=].

Reply:
@domfarolino Any ideas for this?

In the implementation, it fails here:
https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/interest_group/ad_auction_service_impl.cc;l=249;drc=f80633b34538615fcb73515ad8c4bc56a748abfe;bpv=1;bpt=1

Though that seems to maybe return null, rather than actually fail. I think that's a pretty unhelpful way to fail for "out of resources", though it's exceedingly likely to run into this error in real use.

Reply:
So this is the renderer path that would be hit for this case right? https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;l=3342;drc=51e1b713f6da38219910bf8fb93a81262340bf97 It honestly seems like we should just change this to reject (although I'm not sure under what _other_ circumstances we get a null config here). I guess I'd suggest this:

 1. Making the spec match the implementation for now, just in case we never change the implementation at least we have consistency
 2. Filing a bug to change the implementation (and then, spec) to reject in this case, and look into what other cases the "browser returns a null config" path is hit (if any other besides the resource exhaustion one)

Reply:
Right, I think that is the same case as "auction doesn't return a winner".

So I guess I should just resolve |p| with null, and file a crbug to consider changing the means of expressing failure.

Reply:
Modified the spec to resolve |p| to null and return it, and created https://crbug.com/1458047